### PR TITLE
update_agent: add hidden knob to override interactive session check

### PR DESCRIPTION
**Pull request by jlebon**
An issue as old as ~time~Fedora CoreOS is that when you're testing something related to updates, you want Zincati to reboot but because `cosa run` gives you an interactive session and you can't *not* have a session without losing the VM, you're stuck waiting for the 10 minutes grace period to expire.

Just add a hidden knob for this to make that scenario less painful.

The API is pretty much: when you're ready to have Zincati reboot, just `touch /run/zincati/override-interactive-check`.

### CODE REVIEW

**Comment by dustymabe** [comment]
I think this touches on a paper cut I thought would be handled by https://github.com/coreos/zincati/issues/498 when we finally implemented that.

As is it improves the state, but I think users should have a documented way to do this too.

**Review by dustymabe — APPROVED** [review]
LGTM

**Comment by dustymabe** [comment]
> I think this touches on a paper cut I thought would be handled by #498 when we finally implemented that.

for example, #498 could handle the "I'm in an interactive session, an update is available, and I want to ignore update strategy (i.e. if your node isn't configured to update until the weekend)" case.

File: src/update_agent/mod.rs [hunk 1]
```diff
@@ -37,6 +38,10 @@ const DEFAULT_POSTPONEMENT_TIME_SECS: u64 = 60; // 1 minute.
 /// before abandoning a target update.
 const MAX_DEPLOY_ATTEMPTS: u8 = 12;
 
+/// This is undocumented; it's for testing Zincati in e.g. cosa where you have
+/// an interactive session but you don't want to wait for the full timeout.
+const INTERACTIVE_SESSION_OVERRIDE: &str = "/run/zincati/override-interactive-check";
```
Comment (cgwalters) [inline src/update_agent/mod.rs:43]: Seems OK as is, but I personally would have probably leaned towards making this an environment variable and told people to `systemctl edit --runtime zincati` instead.

**Review by cgwalters — APPROVED** [review]

File: src/update_agent/mod.rs (hunk 1 above)
Comment (jlebon) [reply src/update_agent/mod.rs:43]: Yeah, I debated on this a bit. It's just... so much easier to touch a file than to type out a dropin. Though we could also have added a `cosa run --add-ignition zincati-override-interactive-check` to help.

**Comment by jlebon** [comment]
> > I think this touches on a paper cut I thought would be handled by #498 when we finally implemented that.

Yeah, thanks for calling out #498. I agree it'd mostly solve that case, though OTOH it's also a lot more work. I found this useful for debugging https://github.com/coreos/fedora-coreos-tracker/issues/1691 and it took a few minutes to type it out. We can nuke it when #498 happens.

**Pull request merged**